### PR TITLE
Adds the Artifact of Kin as a station trait.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Adds a station trait that makes dynamic pick a single antagonist for roundstart, latejoin, and midround.

## Why It's Good For The Game
I find the image of the station getting invaded by a steady stream of provocateurs amusing.

## Changelog

:cl:
expansion: Very occasionally a few types of antagonists will be chosen to be the only types of antagonists that shows up that shift.
/:cl:

### CODE REVIEW

**Comment by IndieanaJones** [comment]
I really like this.  A lot of station traits don't really do much of anything except offset things at roundstart, but this is the kind of round variation I think the system could really use.

**Comment by Colovorat** [comment]
Shrine of Order moment

**Comment by Y0SH1M4S73R** [comment]
> hmm yes which antags will be the only antags that can spawn during this Impending Doom round?
> Roundstart? Hmm... Thief
> Latejoin? How about..... Thief
> Midround? Let's go with......... Thief

On a more serious note, this trait rolling a heretic-centric round would be *very* interesting.

**Comment by SkeletalElite** [comment]
What happens if the roll is something where 1 roll is hte max allowed like cult, nukies, revs? Will it allow additional rolls of those to occur?

**Comment by TemporalOroboros** [comment]
> What happens if the roll is something where 1 roll is hte max allowed like cult, nukies, revs? Will it allow additional rolls of those to occur?

Given its current state. Yes.

**Comment by TemporalOroboros** [comment]
There is a slight issue that will need to be fixed before merge. If the roundstart picker doesn't roll a high impact ruleset it's possible for latejoin or midround to roll one and lock itself out of spending threat. I'll need to figure out a way to prevent that from happening.

**Comment by TemporalOroboros** [comment]
Basically:
1. Artifact of Kin station trait is enabled.
2. Roundstart picks traitors. All roundstart threat is spent on traitors. Since they are not a high impact ruleset latejoin and midrounds

**Comment by TemporalOroboros** [comment]
Fat fingered close trying to comment. Ignore that.

**Comment by TemporalOroboros** [comment]
That should make midrounds and latejoins picking high impact rulesets not lock themselves out.

**Comment by Mothblocks** [comment]
Labelling DNM until [Dynamic 2022 Part 1](https://github.com/tgstation/tgstation/pull/65997) is merged

File: code/game/gamemodes/dynamic/dynamic.dm
Comment (Mothblocks) [inline code/game/gamemodes/dynamic/dynamic.dm:66]: You need to change the documentation for this to include that it is a bitfield, and then link to the flags it can use.

File: code/game/gamemodes/dynamic/dynamic.dm
Comment (Mothblocks) [inline code/game/gamemodes/dynamic/dynamic.dm:726]: Combine these into one.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Does this respect LONE_RULESET?

This needs to be split out into a datum handler or similar that listens to signals so that we don't have all this code in Dynamic.

Also, depending on how this works, this is not going to function well with Dynamic 2022 at all, since Dynamic 2022's first midround is basically always a sleeper agent.

This needs some marker in the `dynamic.json` log output so I know to completely throw these out when analyzing rounds.

**Comment by Wallemations** [comment]
Soon we'll get Artifact of Command
Choose Your Own Antag adventure

**Comment by D4C-420** [comment]
Artifact of swarms 
Twice the Antags

**Comment by Fikou** [comment]
can we name it something that makes sense to people who didnt play a video game

**Comment by TemporalOroboros** [comment]
I'm honestly going to wait on this until Dynamic 2022 is merged. Then I'll see where I can go with it.

**Comment by Mothblocks** [comment]
Gonna anti-stale this until D2022

**Comment by Mothblocks** [comment]
Artifact of kin PR is that way #66034

**Comment by TemporalOroboros** [comment]
> Artifact of kin PR is that way #66034

I don't think this is doing much good here.

**Comment by Mothblocks** [comment]
God damnit

**Comment by TheSilverNuke** [comment]
> can we name it something that makes sense to people who didnt play a video game

What game is this even from

**Comment by Annihilit** [comment]
> God damnit

Well now you hopefully see how i managed to mess it up in the first place lol

**Comment by private-tristan** [comment]
> What game is this even from


 [Risk of Rain](https://riskofrain.fandom.com/wiki/Kin)

**Comment by TemporalOroboros** [comment]
Alright. Let's give this another go.

**Comment by TemporalOroboros** [comment]
Needs mapping and spriting.

**Comment by TemporalOroboros** [comment]
> > Also, depending on how this works, this is not going to function well with Dynamic 2022 at all, since Dynamic 2022's first midround is basically always a sleeper agent.
> 
> This is still a pretty nasty gotcha I think

Yeah. I've set it to just pick a ruleset for midrounds and then ensure that only that midround can spawn. The result should be less weighted towards sleeper agents than locking in the first picked ruleset. This is still not optimal, since it's possible for it to pick something like nuke ops and then the station has to sit around for the heavy chance to build up high enough for it to roll. Haven't put much thought into how the middle ground would be implemented, but I'm sure there is one.

**Comment by TemporalOroboros** [comment]
On the other hand. Rolling something expensive and waiting for the director to roll it does line up pretty nicely with the RoR experience.

**Pull request closed without merging**
